### PR TITLE
Updates-to-community-API-key-pattern

### PR DESCRIPTION
**Pull request by chloepoulterdesign**
Closes https://github.ibm.com/CDAI-design/pal/issues/263

As part of merging security's Generate an API key pattern with the Carbon Community (CDAI) pattern, we need to make some changes to the documentation.

Many thanks to Shelley Dignum for auditing the patterns. 
FYI @vikkipaterson and @levinson1

#### Changelog

**New**

- section explaining the variety of warnings about API key retrieval that should be provided to users
- section explaining the order that restricted access options should be included in the modal
- section on a full-page API key management page
- section on editing API keys
- guidance that the generated key should appear in its' own modal view, not in the same view as the options form, as the key is the output of a "generation" action which happens after a user has specified the options they desire

**Changed**

- Images in the "Name the API key" section to remove the (confusing) progress indicator
 
**Removed**

- Image containing the destination selector in the "Name the API key" section to avoid confusion with the restricted access section

### CODE REVIEW

**Comment by vikkipaterson** [comment]
This looks great to me Chloe :) Only comment I have is why are these items number 6,7,8 - where is 1-5?
image

**Comment by chloepoulterdesign** [comment]
Good spot Vikki! That's so weird - and when I build locally, it renders 1,2,3!
image 
The numbers are generated when the site is built, so I assume it's an error with Vercel's deployment, and when it's actually pushed live this wouldn't be an issue. 
Can anyone from Carbon confirm that the numbers will appear correctly when this is merged?

File: src/pages/community/patterns/generate-an-api-key/index.mdx [hunk 1]
```diff
@@ -95,19 +98,73 @@ This is particularly useful in instances where a user may have keys for several
...
+  If a key is recoverable in another UI or platform, inform the user and link to
+  instructions for recovering the key.
+</Caption>
+
+</Column>
+</Row>
+
+#### API key cannot be retrieved at a later date
+
+If a user cannot retrieve the API key in future, warn them in the modal. Provide
```
Comment (jillianhowarth) [inline src/pages/community/patterns/generate-an-api-key/index.mdx:157]:
```suggestion
If a user cannot retrieve the API key in the future, warn them in the modal. Provide
```

File: src/pages/community/patterns/generate-an-api-key/index.mdx [hunk 2]
```diff
@@ -118,8 +175,13 @@ A user may want to limit the access an application or service is granted.
 Restricted access allows a user to assign specific access control and
 permissions before the API key is generated.
 
+Restricted access options vary widely and are not specified in this pattern.
+Options should be provided in the modal, **before** the user clicks generate. To
```
Comment (jillianhowarth) [inline src/pages/community/patterns/generate-an-api-key/index.mdx:179]:
```suggestion
Options should be provided in the modal **before** the user clicks generate. To
```

File: src/pages/community/patterns/generate-an-api-key/index.mdx [hunk 3]
```diff
@@ -159,3 +237,49 @@ The user may also need the option to generate a new API key.
 
 </Column>
 </Row>
+
+## API key managment page (Optional)
```
Comment (jillianhowarth) [inline src/pages/community/patterns/generate-an-api-key/index.mdx:241]:
```suggestion
## API key management page (Optional)
```

File: src/pages/community/patterns/generate-an-api-key/index.mdx [hunk 4]
```diff
@@ -28,6 +28,8 @@ The reusable code component can be found in
   <AnchorLink>General access keys</AnchorLink>
   <AnchorLink>Restricted access keys</AnchorLink>
   <AnchorLink>Show the API key in context (optional)</AnchorLink>
+  <AnchorLink>API key management page (Optional)</AnchorLink>
```
Comment (tw15egan) [inline src/pages/community/patterns/generate-an-api-key/index.mdx:31]:
```suggestion
  <AnchorLink>API key management page (optional)</AnchorLink>
```

File: src/pages/community/patterns/generate-an-api-key/index.mdx [hunk 5]
```diff
@@ -159,3 +237,49 @@ The user may also need the option to generate a new API key.
 
 </Column>
 </Row>
+
+## API key management page (Optional)
```
Comment (tw15egan) [inline src/pages/community/patterns/generate-an-api-key/index.mdx:241]:
```suggestion
## API key management page (optional)
```

**Pull request merged**